src/cdk/table/table.ts

```diff
@@ -612,6 +621,7 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     const changes = this._dataDiffer.diff(this._renderRows);
     if (!changes) {
       this._updateNoDataRow();
+      this.dataRowsChange.next();
```

Comment:
Doesn't this call mean that it'll fire the event even if the rows haven't changed?